drivers/include/digit7seg.h

```diff
@@ -0,0 +1,146 @@
+
+#include <inttypes.h>
+
+#include "macros/units.h"
+#include "periph/gpio.h"
+#include "periph/timer.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define DIGIT7SEG_MAX_DIGITS  (4)     /**< Max pin number, can be below or equal */
+#ifndef DIGIT7SEG_TIMER_HZ
+#   define DIGIT7SEG_TIMER_HZ    MHZ(1)  /**< default timer hz */
+#endif
+#ifndef DIGIT7SEG_DELAY
+/**
+ * With 1MHz and a cycle as 2600 we got 2.6ms each shift,
+ * so undetectable frequency by human eyes
+ * */
+#   define DIGIT7SEG_DELAY       (DIGIT7SEG_TIMER_HZ / 2600)
+#endif
+
+/**
+ * @brief   Return codes for @ref digit7seg_init
+ */
+typedef enum {
+    DIGIT7SEG_OK = 0,           /**< All ok */
+    DIGIT7SEG_ERR_DIGITS,       /**< Something went wrong with digits value */
+} digit7seg_error_codes;
+
+/**
+ * @brief   Device initialization parameters, each data_* pin representing
+ *          a segment and dig* representing a digit pin to turn on.
+ * ```
+ *        --a--
+ *       |     |
+ *       f     b
+ *       |     |
+ *        --g--
+ *       |     |
+ *       e     c
+ *       |     |
+ *        --d--    .dp
+ *```
+ */
+typedef struct {
+    gpio_t data_a;  /**< Data input pin */
+    gpio_t data_b;  /**< Data input pin */
+    gpio_t data_c;  /**< Data input pin */
+    gpio_t data_d;  /**< Data input pin */
+    gpio_t data_e;  /**< Data input pin */
+    gpio_t data_f;  /**< Data input pin */
+    gpio_t data_g;  /**< Data input pin */
+    gpio_t data_dp; /**< Data input pin */
+    gpio_t dig1;    /**< Data input pin */
+    gpio_t dig2;    /**< Data input pin */
+    gpio_t dig3;    /**< Data input pin */
+    gpio_t dig4;    /**< Data input pin */
+    tim_t timer;    /**< Timer to use in digit7seg_power* */
+    uint8_t digits; /**< Number of digits */
+} digit7seg_params_t;
+
+/**
+ * @brief   Device descriptor for the DIGIT7SEG
+ */
+typedef struct {
+    digit7seg_params_t params; /**< Device parameters */
+    uint32_t value;            /**< Binary value to display */
+    uint8_t current_digit;     /**< Displayed digit */
+} digit7seg_t;
+
+/**
+ * @brief   Initialize the given DIGIT7SEG
+ *
+ * @param[out] dev      DIGIT7SEG device descriptor to initialize
+ * @param[in]  params   Device parameters to use
+ *
+ * @return  0 on success or negative error code, see #digit7seg_error_codes
+ */
+int digit7seg_init(digit7seg_t *dev, const digit7seg_params_t *params);
+
+/**
+ * @brief   Set the value for every digit
+ *
+ * @param[in] dev       Initialized device descriptor of DIGIT7SEG device
+ * @param[in] value     the value as an uint32_t
+ *
+ */
+void digit7seg_set_all_value(digit7seg_t *dev, uint32_t value);
+
+/**
+ * @brief   Set the value for one digit
+ *
+ * @param[in] dev       Initialized device descriptor of DIGIT7SEG device
+ * @param[in] index     the digit number between [0, DIGIT7SEG_MAX_DIGITS[
+ * @param[in] value     the value as an uint8_t, a is equal to the first bit
+ *                       b the second.... dp the 8th bit.
```

Comment:
As a user, I would expect the driver to do the number to 7-segment-bit conversion for me.

Reply:
Fair, I will work on it !